<launch> 
  <!-- ARGUMENT DEFINITIONS -->
  <arg name="file" default="default.mat" />
  <arg name="dir" default="$(find robot_simulator)/data/" />
  <arg name="vis" default="true" />
  <arg name="index" value="1" />
  <arg name="freq" default="10.0" />
  <arg name="window" default="20" />
  <arg name="tf" default="10.0" /> 
  <arg name="noise" default="0.0" />
  
  <!-- load serial node dictionary -->
  <rosparam file="$(find serial_node)/data/planar_cl_controller_serial_device_dict.xml" 
	    command="load"/>

  <!-- start up the kinect and both planar trackers -->
  <include file="$(find objecttracker_nu)/launch/nodelet_planar_system.launch" >
    <arg name="input" value="points_throttled" />
  </include>

  <group ns="robot_1" >
    <arg name="rn" value="robot_1" />
  </group>

  <!-- set all parameters for this namespace -->
  <param name="robot_index" value="$(arg index)" />
  <param name="window_length" value="$(arg window)" type="int" />
  <param name="controller_freq" value="$(arg freq)" type="double" />
  <param name="time_final" value="$(arg tf)" type="double" />
  <param name="/simulator_noise" value="$(arg noise)" type="double" />
    
  <!-- slow kinect data to a lower frequency -->
  <node pkg="topic_tools" type="throttle" name="kinect_throttler"
	args="messages /camera/depth/points $(arg freq)
	      /camera/depth/points_throttled" 
	respawn="true"/>

  <!-- now I want to start up the controller/ filter, and the
       coordinator for the system -->
  <node pkg="receding_planar_sys" name="receding_controller"
	type="receding_controller.py" output="screen" respawn="true" />
  <node pkg="receding_planar_sys" name="coordinator"
	type="receding_coordinator" output="screen" respawn="true" />

  <!-- start serial node -->
  <node pkg="serial_node" type="multi_serial_topic" name="serial_node"
    	output="log" respawn="true" />
    
  <!-- start up the keyboard node -->
  <node pkg="keyboard_node" type="multi_keyboard_node"
	name="keyboard_interface" output="screen" respawn="true"
	launch-prefix="xterm -e" />

  <!-- start up rviz unless visualization is set to false -->
  <group if="$(arg vis)">
    <node pkg="rviz" type="rviz" respawn="false" name="rviz"
	  args="-d $(find robot_simulator)/launch/controller.vcg" />
  </group>
  

</launch>
